File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Deque<TreeNode> queue = new ArrayDeque<>();
        List<List<Integer>> res = new ArrayList<>();

        if (root == null)
            return res;
        int i = 0, l = 0, r = 1;
        queue.addLast(root);
        while (queue.size() != 0) {
            List<Integer> list = new ArrayList<>();
            while (l < r) {
                TreeNode node = queue.pollFirst();
                l++;
                if (node.left != null) {
                    queue.addLast(node.left);
                }
                if (node.right != null) {
                    queue.addLast(node.right);
                }
                list.add(node.val);
            }
            res.add(list);
            l = 0;
            r = queue.size();
        }
        return res;
    }
}

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    postorderTraversal(root: TreeNode | null): number[] {
        let res = []

        this.traverse(root, res)

        return res
    }

    traverse(node: TreeNode | null, res: number[]) {
        if (node == null) return
        this.traverse(node.left, res)
        this.traverse(node.right, res)
        res.push(node.val)
    }
}
